src/AbstractExtendedPdo.php

```diff
@@ -634,20 +636,18 @@ public function prepareWithValues(string $statement, array $values = []): PDOSta
      *
      * @param string $query The SQL statement to prepare and execute.
      *
-     * @param int|null $fetchMode
-     *
-     * @param mixed ...$fetch_mode_args Optional fetch-related parameters.
+     * @param mixed ...$fetch Optional fetch-related parameters.
      *
      * @return PDOStatement
      *
      * @see http://php.net/manual/en/pdo.query.php
      *
      */
-    public function query(string $query, ?int $fetchMode = null, mixed ...$fetch_mode_args): PDOStatement
+    public function query(string $query, ...$fetch): PDOStatement
```

Comment:
This is different from 5.x.
But it seems 3.x is the same.
https://github.com/auraphp/Aura.Sql/blob/f19b1aa810b35d86ad55562876d967ef3f9a97b7/src/AbstractExtendedPdo.php#L645

Is this okay?

Reply:
I had to revert to that to avoid type signature conflicts in either 8.1 or earlier versions. Specifically, the default fetch mode per docs is a constant that evaluates to 19, but that constant is out of range for PDO itself rather than evaluating to whatever the default on-create waa. By contrast, defaulting to null and passing that through failed type checks as well. The only thing that worked across the board was the 3.x version.

If you swap the signature for the 5.x version, or the 5.x version without typehints that are only available in 8.0+, you'll see the errors that I saw this morning :)

Reply:
Okay, I understand 4.x should have the method signature.